packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx

```diff
@@ -227,6 +229,23 @@ class Dialog extends AutoControlledComponent<WithAsProp<DialogProps>, DialogStat
     }
   };
 
+  /**
+   * It's needed to control after state changes because self controlled dialogs
+   */
+  componentDidUpdate() {
+    const openedDialogs = +this.context.target.body.getAttribute(Dialog.dialogsCounterAttribute) || 0;
+    if (this.state.open) {
```

Comment:
This will occur for every dialog rerender

Reply:
Yes, on the dialog is open/closed based in the changes of state. So where would you suggest to place this logic if not in the `componentDidUpdate`?

Reply:
`Dialog` can be rerendered without changes in the state, for example its parent can produce a render 

In current logic:
```
render ({ state: open }) // disableBodyScroll
render ({ state: open }) // disableBodyScroll
render ({ state: open }) // disableBodyScroll
// still will executed
```

```suggestion
    if (!prevState.open && this.state.open) { 
```

In this case it will executed what it will really become opened. `prevState` is a second param in [`didUpdate`](https://reactjs.org/docs/react-component.html#componentdidupdate)